src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,59 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int STORAGE_MAX_SIZE = 10;
+    private Object[][] items = new Object[STORAGE_MAX_SIZE][2];
+    private int size = 0;
+
+    public StorageImpl() {
+    }
+
     @Override
     public void put(K key, V value) {
+        int existingKey = -1;
+        int firstFreeSpot = -1;
+        for (int i = 0; i < STORAGE_MAX_SIZE; i++) {
```

Comment:
```suggestion
        for (int i = 0; i < size; i++) {
```
фактично нам достатньо ітеруватись по заповнених комірках тільки